tests/PHPUnit/Infrastructure/Service/Precondition/NoLinksExistOnWindowsFunctionalTest.php

```diff
@@ -0,0 +1,159 @@
+
+        self::assertTrue($isFulfilled, 'Passed with no links in the codebase.');
+    }
+
+    /**
+     * @covers ::exitEarly
+     * @covers ::findFiles
+     * @covers ::getDefaultUnfulfilledStatusMessage
+     * @covers ::getUnfulfilledStatusMessage
+     * @covers ::isFulfilled
+     * @covers ::isSupportedLink
+     *
+     * @dataProvider providerUnfulfilled
+     */
+    public function testUnfulfilled(array $symlinks, array $hardLinks): void
+    {
+        // This test is host-sensitive and can only be run on Windows.
+        if (!self::isWindows()) {
+            $this->expectNotToPerformAssertions();
+
+            return;
+        }
+
+        $baseDir = PathFactory::create(self::ACTIVE_DIR);
+        $link = PathFactory::create('link.txt', $baseDir)->resolve();
+        $target = PathFactory::create('target.txt', $baseDir)->resolve();
+
+        $this->expectException(PreconditionException::class);
+        $this->expectExceptionMessage(sprintf(
+            'The active directory at "%s" contains links, which is not supported on Windows. The first one is "%s".',
+            $baseDir->resolve(),
+            $link,
+        ));
+
+        touch($target);
+        self::createSymlinks($baseDir->resolve(), $symlinks);
+        self::createHardlinks($baseDir->resolve(), $hardLinks);
+        $sut = $this->createSut();
+
+        $isFulfilled = $sut->isFulfilled($this->activeDir, $this->stagingDir);
+
+        self::assertFalse($isFulfilled, 'Rejected link on Windows.');
+
+        $sut->assertIsFulfilled($this->activeDir, $this->stagingDir);
+    }
+
+    public function providerUnfulfilled(): array
+    {
+        return [
+            'Contains symlink' => [
+                'symlinks' => ['link.txt' => 'target.txt'],
+                'hardLinks' => [],
+            ],
+            'Contains hard link' => [
+                'symlinks' => [],
+                'hardLinks' => ['link.txt' => 'target.txt'],
+            ],
+        ];
+    }
+
+    /**
+     * @covers ::exitEarly
+     * @covers ::findFiles
+     * @covers ::getDefaultUnfulfilledStatusMessage
+     * @covers ::isFulfilled
+     *
+     * @uses \PhpTuf\ComposerStager\Infrastructure\Service\Filesystem\Filesystem
+     * @uses \PhpTuf\ComposerStager\Infrastructure\Service\Precondition\AbstractPrecondition
+     *
+     * @dataProvider providerDirectoryDoesNotExist
+     */
+    public function testDirectoryDoesNotExist(string $activeDir, string $stagingDir): void
+    {
+        $this->doTestDirectoryDoesNotExist($activeDir, $stagingDir);
+    }
+
+    /**
+     * @covers ::exitEarly
+     * @covers ::getDefaultUnfulfilledStatusMessage
+     * @covers ::isFulfilled
+     *
+     * @dataProvider providerExclusions
+     */
+    public function testExclusions(array $links, array $exclusions, bool $shouldBeFulfilled): void
+    {
+        if (!self::isWindows()) {
+            $this->expectNotToPerformAssertions();
+
+            return;
+        }
```

Comment:
Apparently it _needs_ a comment, because it _only_ runs on Windows. 😝 I'll add the same comment used earlier in the file:

```
// This test is host-sensitive and can only be run on Windows.
```

Reply:
Thanks! 😊 Much clearer! 👍